controllers/containerjfr_controller.go

```diff
@@ -328,7 +361,21 @@ func (r *ContainerJFRReconciler) createService(ctx context.Context, controller *
 			InsecureEdgeTerminationPolicy: openshiftv1.InsecureEdgeTerminationPolicyRedirect,
 		}
 	}
-	return r.createRouteForService(controller, svc, *exposePort, tlsConfig)
+	if r.IsOpenShift {
+		return r.createRouteForService(controller, svc, *exposePort, tlsConfig)
+	} else {
+		if controller.Spec.NetworkOptions == nil {
+			return &url.URL{Host: ""}, nil
```

Comment:
I would return `nil, nil` here, then add a nil-check when this URL is to be used. e.g. Only add the respective environment variables (`CONTAINER_JFR_WEB_HOST`, `GRAFANA_DASHBOARD_URL`, etc.) if the URLs are non-nil.